src/NUnit.Xml.TestLogger/NUnitXmlTestLogger.cs

```diff
@@ -0,0 +1,501 @@
+            element.SetAttributeValue("passed", testSuites.Sum(x => (int) x.Attribute("passed")));
+            element.SetAttributeValue("failed", testSuites.Sum(x => (int) x.Attribute("failed")));
+            element.SetAttributeValue("skipped", testSuites.Sum(x => (int) x.Attribute("skipped")));
+
+            element.SetAttributeValue("run-date", localStartTime.ToString("yyyy-MM-dd", CultureInfo.InvariantCulture));
+            element.SetAttributeValue("run-time", localStartTime.ToString("HH:mm:ss", CultureInfo.InvariantCulture));
+
+            return element;
+        }
+
+        private static string DetermineResultFromDescendants(XElement element, string descendantElementName)
+        {
+            string resultString = ResultStatusPassed;
+            foreach (var x in element.Descendants(descendantElementName))
+            {
+                if (ResultStatusFailed.Equals((string) x.Attribute("result"), StringComparison.OrdinalIgnoreCase))
+                {
+                    resultString = ResultStatusFailed;
+                    break;
+                }
+            }
+            return resultString;
+        }
+
+        private XElement CreateTestSuiteElement(IGrouping<string, TestResultInfo> resultsByAssembly)
+        {
+            var assemblyPath = resultsByAssembly.Key;
+
+            var collections = from resultsInAssembly in resultsByAssembly
+                group resultsInAssembly by resultsInAssembly.Type
+                into resultsByType
+                orderby resultsByType.Key
+                select CreateTestSuite(resultsByType);
+
+            int total = 0;
+            int passed = 0;
+            int failed = 0;
+            int skipped = 0;
+            int errors = 0;
+            var time = TimeSpan.Zero;
+
+            var element = new XElement("test-suite");
+            element.SetAttributeValue("type", "Assembly");
+
+            XElement errorsElement = new XElement("errors");
+            element.Add(errorsElement);
+
+            foreach (var collection in collections)
+            {
+                total += collection.total;
+                passed += collection.passed;
+                failed += collection.failed;
+                skipped += collection.skipped;
+                errors += collection.error;
+                time += collection.time;
+
+                element.Add(collection.element);
+            }
+
+            element.SetAttributeValue("name", Path.GetFileName(assemblyPath));
+            element.SetAttributeValue("fullname", assemblyPath);
+
+            element.SetAttributeValue("total", total);
+            element.SetAttributeValue("passed", passed);
+            element.SetAttributeValue("failed", failed);
+            element.SetAttributeValue("skipped", skipped);
+            element.SetAttributeValue("time", time.TotalSeconds.ToString("N3", CultureInfo.InvariantCulture));
+            element.SetAttributeValue("errors", errors);
```

Comment:
There is no `result` on assembly level.
This is the output we are getting:
```xml
<test-suite type="Assembly" name="NUnit.Xml.TestLogger.NetCore.Tests.dll" fullname="D:\enlistments\LoggerExtensions\test\NUnit.Xml.TestLogger.NetCore.Tests\bin\Debug\netcoreapp1.0\NUnit.Xml.TestLogger.NetCore.Tests.dll" total="21" passed="7" failed="7" skipped="7" time="0.501" errors="0">
```

Reply:
Added result for test-suite.